SeminKang/JS100제/q50.md

```diff
@@ -0,0 +1,50 @@
+버블정렬은 두 인접한 원소를 검사하여 정렬하는 방법을 말합니다. 시간 복잡도는 느리지만 코드가 단순하기 때문에 자주 사용됩니다.
+
+[버블정렬 이미지 참고](https://www.notion.so/50-bb22106895db4766b2517b0f61a8cc2d)
+## 내 코드
+그럼 내가 한건 버블 정렬이 아닌가? 오옹 [8, 9, 10, 1, 2] 돌리니까 [8, 1, 2, 9, 10]이 나옴..
+
+j가 0부터 시작하면서 모든걸 다 검토하는게 아니라서 그런듯!
+```js
+function bubble(arr) {
+  let result = arr.slice(); 
+
+  for (let i = 0; i < result.length - 1; i++) {
+    for (let j = i; j < result.length - 1; j++) {
```

Comment:
for (let j = i; j < result.length - 1; j++) → for (let j = i; j < result.length - i; j++)

https://www.toptal.com/developers/sorting-algorithms